File: deploy/windows/src/haxor/component/Renderer.cpp
#include <hxcpp.h>

#ifndef INCLUDED_haxor_component_Behaviour
#include <haxor/component/Behaviour.h>
#endif
#ifndef INCLUDED_haxor_component_Component
#include <haxor/component/Component.h>
#endif
#ifndef INCLUDED_haxor_component_Renderer
#include <haxor/component/Renderer.h>
#endif
#ifndef INCLUDED_haxor_context_EngineContext
#include <haxor/context/EngineContext.h>
#endif
#ifndef INCLUDED_haxor_context_RendererContext
#include <haxor/context/RendererContext.h>
#endif
#ifndef INCLUDED_haxor_core_IDisposable
#include <haxor/core/IDisposable.h>
#endif
#ifndef INCLUDED_haxor_core_Resource
#include <haxor/core/Resource.h>
#endif
#ifndef INCLUDED_haxor_graphics_material_Material
#include <haxor/graphics/material/Material.h>
#endif
namespace haxor{
namespace component{

Void Renderer_obj::__construct(::String __o_p_name)
{
HX_STACK_FRAME("haxor.component.Renderer","new",0x83675ed2,"haxor.component.Renderer.new","haxor/component/Renderer.hx",13,0x7c4f98dc)
HX_STACK_THIS(this)
HX_STACK_ARG(__o_p_name,"p_name")
::String p_name = __o_p_name.Default(HX_CSTRING(""));
{
	HX_STACK_LINE(44)
	this->m_has_mesh = false;
	HX_STACK_LINE(13)
	super::__construct(p_name);
}
;
	return null();
}

//Renderer_obj::~Renderer_obj() { }

Dynamic Renderer_obj::__CreateEmpty() { return  new Renderer_obj; }
hx::ObjectPtr< Renderer_obj > Renderer_obj::__new(::String __o_p_name)
{  hx::ObjectPtr< Renderer_obj > result = new Renderer_obj();
	result->__construct(__o_p_name);
	return result;}

Dynamic Renderer_obj::__Create(hx::DynamicArray inArgs)
{  hx::ObjectPtr< Renderer_obj > result = new Renderer_obj();
	result->__construct(inArgs[0]);
	return result;}

::haxor::graphics::material::Material Renderer_obj::get_material( ){
	HX_STACK_FRAME("haxor.component.Renderer","get_material",0x97f323fe,"haxor.component.Renderer.get_material","haxor/component/Renderer.hx",21,0x7c4f98dc)
	HX_STACK_THIS(this)
	HX_STACK_LINE(21)
	return this->m_material;
}


HX_DEFINE_DYNAMIC_FUNC0(Renderer_obj,get_material,return )

::haxor::graphics::material::Material Renderer_obj::set_material( ::haxor::graphics::material::Material v){
	HX_STACK_FRAME("haxor.component.Renderer","set_material",0xacec4772,"haxor.component.Renderer.set_material","haxor/component/Renderer.hx",23,0x7c4f98dc)
	HX_STACK_THIS(this)
	HX_STACK_ARG(v,"v")
	HX_STACK_LINE(24)
	if (((this->m_material == v))){
		HX_STACK_LINE(24)
		return v;
	}
	HX_STACK_LINE(25)
	this->m_material = v;
	HX_STACK_LINE(26)
	if (((v == null()))){
		HX_STACK_LINE(26)
		this->m_last_queue = (int)-1;
	}
	else{
		HX_STACK_LINE(26)
		this->m_last_queue = v->queue;
	}
	HX_STACK_LINE(27)
	::haxor::context::EngineContext_obj::renderer->OnMaterialChange(hx::ObjectPtr<OBJ_>(this));
	HX_STACK_LINE(28)
	return v;
}


HX_DEFINE_DYNAMIC_FUNC1(Renderer_obj,set_material,return )

bool Renderer_obj::get_visible( ){
	HX_STACK_FRAME("haxor.component.Renderer","get_visible",0x4ef128fb,"haxor.component.Renderer.get_visible","haxor/component/Renderer.hx",36,0x7c4f98dc)
	HX_STACK_THIS(this)
	HX_STACK_LINE(36)
	return (bool(this->m_visible) && bool(!(this->m_culled)));
}


HX_DEFINE_DYNAMIC_FUNC0(Renderer_obj,get_visible,return )

bool Renderer_obj::set_visible( bool v){
	HX_STACK_FRAME("haxor.component.Renderer","set_visible",0x595e3007,"haxor.component.Renderer.set_visible","haxor/component/Renderer.hx",37,0x7c4f98dc)
	HX_STACK_THIS(this)
	HX_STACK_ARG(v,"v")
	HX_STACK_LINE(37)
	this->m_visible = v;
	HX_STACK_LINE(37)
	return v;
}


HX_DEFINE_DYNAMIC_FUNC1(Renderer_obj,set_visible,return )

Void Renderer_obj::OnBuild( ){
{
		HX_STACK_FRAME("haxor.component.Renderer","OnBuild",0xb11a50e1,"haxor.component.Renderer.OnBuild","haxor/component/Renderer.hx",55,0x7c4f98dc)
		HX_STACK_THIS(this)
		HX_STACK_LINE(56)
		this->super::OnBuild();
		HX_STACK_LINE(57)
		this->m_last_queue = (int)-1;
		HX_STACK_LINE(58)
		this->m_visible = true;
		HX_STACK_LINE(59)
		this->m_culled = false;
		HX_STACK_LINE(60)
		::haxor::context::EngineContext_obj::renderer->Create(hx::ObjectPtr<OBJ_>(this));
	}
return null();
}


Void Renderer_obj::OnRender( ){
{
		HX_STACK_FRAME("haxor.component.Renderer","OnRender",0x7880f243,"haxor.component.Renderer.OnRender","haxor/component/Renderer.hx",67,0x7c4f98dc)
		HX_STACK_THIS(this)
		HX_STACK_LINE(67)
		if (((this->m_material != null()))){
			HX_STACK_LINE(68)
			if (((this->m_last_queue != this->m_material->queue))){
				HX_STACK_LINE(70)
				this->m_last_queue = this->m_material->queue;
				HX_STACK_LINE(71)
				::haxor::context::EngineContext_obj::renderer->OnMaterialChange(hx::ObjectPtr<OBJ_>(this));
			}
		}
	}
return null();
}


HX_DEFINE_DYNAMIC_FUNC0(Renderer_obj,OnRender,(void))

Void Renderer_obj::UpdateCulling( ){
{
		HX_STACK_FRAME("haxor.component.Renderer","UpdateCulling",0x62436839,"haxor.component.Renderer.UpdateCulling","haxor/component/Renderer.hx",79,0x7c4f98dc)
		HX_STACK_THIS(this)
		HX_STACK_LINE(82)
		bool v0 = this->m_culled;		HX_STACK_VAR(v0,"v0");
		HX_STACK_LINE(83)
		bool v1 = this->CheckCulling();		HX_STACK_VAR(v1,"v1");
		HX_STACK_LINE(84)
		this->m_culled = v1;
		HX_STACK_LINE(86)
		if (((v0 != v1))){
			HX_STACK_LINE(88)
			::haxor::context::EngineContext_obj::renderer->OnVisibilityChange(hx::ObjectPtr<OBJ_>(this),!(v1));
		}
	}
return null();
}


HX_DEFINE_DYNAMIC_FUNC0(Renderer_obj,UpdateCulling,(void))

bool Renderer_obj::CheckCulling( ){
	HX_STACK_FRAME("haxor.component.Renderer","CheckCulling",0xfb54f9f6,"haxor.component.Renderer.CheckCulling","haxor/component/Renderer.hx",95,0x7c4f98dc)
	HX_STACK_THIS(this)
	HX_STACK_LINE(95)
	return false;
}


HX_DEFINE_DYNAMIC_FUNC0(Renderer_obj,CheckCulling,return )

Void Renderer_obj::OnDestroy( ){
{
		HX_STACK_FRAME("haxor.component.Renderer","OnDestroy",0x9dc4b56d,"haxor.component.Renderer.OnDestroy","haxor/component/Renderer.hx",102,0x7c4f98dc)
		HX_STACK_THIS(this)
		HX_STACK_LINE(102)
		::haxor::context::EngineContext_obj::renderer->Destroy(hx::ObjectPtr<OBJ_>(this));
	}
return null();
}



Renderer_obj::Renderer_obj()
{
}

void Renderer_obj::__Mark(HX_MARK_PARAMS)
{
	HX_MARK_BEGIN_CLASS(Renderer);
	HX_MARK_MEMBER_NAME(m_material,"m_material");
	HX_MARK_MEMBER_NAME(m_visible,"m_visible");
	HX_MARK_MEMBER_NAME(m_culled,"m_culled");
	HX_MARK_MEMBER_NAME(m_has_mesh,"m_has_mesh");
	HX_MARK_MEMBER_NAME(m_last_queue,"m_last_queue");
	::haxor::component::Component_obj::__Mark(HX_MARK_ARG);
	HX_MARK_END_CLASS();
}

void Renderer_obj::__Visit(HX_VISIT_PARAMS)
{
	HX_VISIT_MEMBER_NAME(m_material,"m_material");
	HX_VISIT_MEMBER_NAME(m_visible,"m_visible");
	HX_VISIT_MEMBER_NAME(m_culled,"m_culled");
	HX_VISIT_MEMBER_NAME(m_has_mesh,"m_has_mesh");
	HX_VISIT_MEMBER_NAME(m_last_queue,"m_last_queue");
	::haxor::component::Component_obj::__Visit(HX_VISIT_ARG);
}

Dynamic Renderer_obj::__Field(const ::String &inName,bool inCallProp)
{
	switch(inName.length) {
	case 7:
		if (HX_FIELD_EQ(inName,"visible") ) { return get_visible(); }
		if (HX_FIELD_EQ(inName,"OnBuild") ) { return OnBuild_dyn(); }
		break;
	case 8:
		if (HX_FIELD_EQ(inName,"material") ) { return get_material(); }
		if (HX_FIELD_EQ(inName,"m_culled") ) { return m_culled; }
		if (HX_FIELD_EQ(inName,"OnRender") ) { return OnRender_dyn(); }
		break;
	case 9:
		if (HX_FIELD_EQ(inName,"m_visible") ) { return m_visible; }
		if (HX_FIELD_EQ(inName,"OnDestroy") ) { return OnDestroy_dyn(); }
		break;
	case 10:
		if (HX_FIELD_EQ(inName,"m_material") ) { return m_material; }
		if (HX_FIELD_EQ(inName,"m_has_mesh") ) { return m_has_mesh; }
		break;
	case 11:
		if (HX_FIELD_EQ(inName,"get_visible") ) { return get_visible_dyn(); }
		if (HX_FIELD_EQ(inName,"set_visible") ) { return set_visible_dyn(); }
		break;
	case 12:
		if (HX_FIELD_EQ(inName,"get_material") ) { return get_material_dyn(); }
		if (HX_FIELD_EQ(inName,"set_material") ) { return set_material_dyn(); }
		if (HX_FIELD_EQ(inName,"m_last_queue") ) { return m_last_queue; }
		if (HX_FIELD_EQ(inName,"CheckCulling") ) { return CheckCulling_dyn(); }
		break;
	case 13:
		if (HX_FIELD_EQ(inName,"UpdateCulling") ) { return UpdateCulling_dyn(); }
	}
	return super::__Field(inName,inCallProp);
}

Dynamic Renderer_obj::__SetField(const ::String &inName,const Dynamic &inValue,bool inCallProp)
{
	switch(inName.length) {
	case 8:
		if (HX_FIELD_EQ(inName,"material") ) { return set_material(inValue); }
		if (HX_FIELD_EQ(inName,"m_culled") ) { m_culled=inValue.Cast< bool >(); return inValue; }
		break;
	case 9:
		if (HX_FIELD_EQ(inName,"m_visible") ) { m_visible=inValue.Cast< bool >(); return inValue; }
		break;
	case 10:
		if (HX_FIELD_EQ(inName,"m_material") ) { m_material=inValue.Cast< ::haxor::graphics::material::Material >(); return inValue; }
		if (HX_FIELD_EQ(inName,"m_has_mesh") ) { m_has_mesh=inValue.Cast< bool >(); return inValue; }
		break;
	case 12:
		if (HX_FIELD_EQ(inName,"m_last_queue") ) { m_last_queue=inValue.Cast< int >(); return inValue; }
	}
	return super::__SetField(inName,inValue,inCallProp);
}

void Renderer_obj::__GetFields(Array< ::String> &outFields)
{
	outFields->push(HX_CSTRING("material"));
	outFields->push(HX_CSTRING("m_material"));
	outFields->push(HX_CSTRING("visible"));
	outFields->push(HX_CSTRING("m_visible"));
	outFields->push(HX_CSTRING("m_culled"));
	outFields->push(HX_CSTRING("m_has_mesh"));
	outFields->push(HX_CSTRING("m_last_queue"));
	super::__GetFields(outFields);
};

static ::String sStaticFields[] = {
	String(null()) };

#if HXCPP_SCRIPTABLE
static hx::StorageInfo sMemberStorageInfo[] = {
	{hx::fsObject /*::haxor::graphics::material::Material*/ ,(int)offsetof(Renderer_obj,m_material),HX_CSTRING("m_material")},
	{hx::fsBool,(int)offsetof(Renderer_obj,m_visible),HX_CSTRING("m_visible")},
	{hx::fsBool,(int)offsetof(Renderer_obj,m_culled),HX_CSTRING("m_culled")},
	{hx::fsBool,(int)offsetof(Renderer_obj,m_has_mesh),HX_CSTRING("m_has_mesh")},
	{hx::fsInt,(int)offsetof(Renderer_obj,m_last_queue),HX_CSTRING("m_last_queue")},
	{ hx::fsUnknown, 0, null()}
};
#endif

static ::String sMemberFields[] = {
	HX_CSTRING("get_material"),
	HX_CSTRING("set_material"),
	HX_CSTRING("m_material"),
	HX_CSTRING("get_visible"),
	HX_CSTRING("set_visible"),
	HX_CSTRING("m_visible"),
	HX_CSTRING("m_culled"),
	HX_CSTRING("m_has_mesh"),
	HX_CSTRING("m_last_queue"),
	HX_CSTRING("OnBuild"),
	HX_CSTRING("OnRender"),
	HX_CSTRING("UpdateCulling"),
	HX_CSTRING("CheckCulling"),
	HX_CSTRING("OnDestroy"),
	String(null()) };

static void sMarkStatics(HX_MARK_PARAMS) {
	HX_MARK_MEMBER_NAME(Renderer_obj::__mClass,"__mClass");
};

#ifdef HXCPP_VISIT_ALLOCS
static void sVisitStatics(HX_VISIT_PARAMS) {
	HX_VISIT_MEMBER_NAME(Renderer_obj::__mClass,"__mClass");
};

#endif

Class Renderer_obj::__mClass;

void Renderer_obj::__register()
{
	hx::Static(__mClass) = hx::RegisterClass(HX_CSTRING("haxor.component.Renderer"), hx::TCanCast< Renderer_obj> ,sStaticFields,sMemberFields,
	&__CreateEmpty, &__Create,
	&super::__SGetClass(), 0, sMarkStatics
#ifdef HXCPP_VISIT_ALLOCS
    , sVisitStatics
#endif
#ifdef HXCPP_SCRIPTABLE
    , sMemberStorageInfo
#endif
);
}

void Renderer_obj::__boot()
{
}

} // end namespace haxor
} // end namespace component

File: deploy/android/src/haxor/component/Renderer.java
package haxor.component;
import haxe.root.*;

@SuppressWarnings(value={"rawtypes", "unchecked"})
public  class Renderer extends haxor.component.Behaviour
{
	public    Renderer(haxe.lang.EmptyObject empty)
	{
		super(((haxe.lang.EmptyObject) (haxe.lang.EmptyObject.EMPTY) ));
	}
	
	
	public    Renderer(java.lang.String p_name)
	{
		super(((haxe.lang.EmptyObject) (haxe.lang.EmptyObject.EMPTY) ));
		haxor.component.Renderer.__hx_ctor_haxor_component_Renderer(this, p_name);
	}
	
	
	public static   void __hx_ctor_haxor_component_Renderer(haxor.component.Renderer __temp_me38, java.lang.String p_name)
	{
		haxor.component.Behaviour.__hx_ctor_haxor_component_Behaviour(__temp_me38, p_name);
		__temp_me38.m_has_mesh = false;
	}
	
	
	public static   java.lang.Object __hx_createEmpty()
	{
		return new haxor.component.Renderer(((haxe.lang.EmptyObject) (haxe.lang.EmptyObject.EMPTY) ));
	}
	
	
	public static   java.lang.Object __hx_create(haxe.root.Array arr)
	{
		return new haxor.component.Renderer(haxe.lang.Runtime.toString(arr.__get(0)));
	}
	
	
	
	
	public final   haxor.graphics.material.Material get_material()
	{
		return this.m_material;
	}
	
	
	public   haxor.graphics.material.Material set_material(haxor.graphics.material.Material v)
	{
		if (( this.m_material == v )) 
		{
			return v;
		}
		
		this.m_material = v;
		if (( v == null )) 
		{
			this.m_last_queue = -1;
		}
		 else 
		{
			this.m_last_queue = v.queue;
		}
		
		haxor.context.EngineContext.renderer.OnMaterialChange(this);
		return v;
	}
	
	
	public  haxor.graphics.material.Material m_material;
	
	
	
	public final   boolean get_visible()
	{
		return ( this.m_visible &&  ! (this.m_culled)  );
	}
	
	
	public   boolean set_visible(boolean v)
	{
		this.m_visible = v;
		return v;
	}
	
	
	public  boolean m_visible;
	
	public  boolean m_culled;
	
	public  boolean m_has_mesh;
	
	public  int m_last_queue;
	
	@Override public   void OnBuild()
	{
		super.OnBuild();
		this.m_last_queue = -1;
		this.m_visible = true;
		this.m_culled = false;
		haxor.context.EngineContext.renderer.Create(this);
	}
	
	
	public   void OnRender()
	{
		if (( this.m_material != null )) 
		{
			if (( this.m_last_queue != this.m_material.queue )) 
			{
				this.m_last_queue = this.m_material.queue;
				haxor.context.EngineContext.renderer.OnMaterialChange(this);
			}
			
		}
		
	}
	
	
	public   void UpdateCulling()
	{
		boolean v0 = this.m_culled;
		boolean v1 = this.CheckCulling();
		this.m_culled = v1;
		if (( v0 != v1 )) 
		{
			haxor.context.EngineContext.renderer.OnVisibilityChange(this,  ! (v1) );
		}
		
	}
	
	
	public   boolean CheckCulling()
	{
		return false;
	}
	
	
	@Override public   void OnDestroy()
	{
		haxor.context.EngineContext.renderer.Destroy(this);
	}
	
	
	@Override public   double __hx_setField_f(java.lang.String field, double value, boolean handleProperties)
	{
		{
			boolean __temp_executeDef634 = true;
			switch (field.hashCode())
			{
				case 259065146:
				{
					if (field.equals("m_last_queue")) 
					{
						__temp_executeDef634 = false;
						this.m_last_queue = ((int) (value) );
						return value;
					}
					
					break;
				}
				
				
			}
			
			if (__temp_executeDef634) 
			{
				return super.__hx_setField_f(field, value, handleProperties);
			}
			 else 
			{
				throw null;
			}
			
		}
		
	}
	
	
	@Override public   java.lang.Object __hx_setField(java.lang.String field, java.lang.Object value, boolean handleProperties)
	{
		{
			boolean __temp_executeDef635 = true;
			switch (field.hashCode())
			{
				case 259065146:
				{
					if (field.equals("m_last_queue")) 
					{
						__temp_executeDef635 = false;
						this.m_last_queue = ((int) (haxe.lang.Runtime.toInt(value)) );
						return value;
					}
					
					break;
				}
				
				
				case 299066663:
				{
					if (field.equals("material")) 
					{
						__temp_executeDef635 = false;
						this.set_material(((haxor.graphics.material.Material) (value) ));
						return value;
					}
					
					break;
				}
				
				
				case 286009348:
				{
					if (field.equals("m_has_mesh")) 
					{
						__temp_executeDef635 = false;
						this.m_has_mesh = haxe.lang.Runtime.toBool(value);
						return value;
					}
					
					break;
				}
				
				
				case 444448953:
				{
					if (field.equals("m_material")) 
					{
						__temp_executeDef635 = false;
						this.m_material = ((haxor.graphics.material.Material) (value) );
						return value;
					}
					
					break;
				}
				
				
				case -1948035677:
				{
					if (field.equals("m_culled")) 
					{
						__temp_executeDef635 = false;
						this.m_culled = haxe.lang.Runtime.toBool(value);
						return value;
					}
					
					break;
				}
				
				
				case -914040160:
				{
					if (field.equals("m_visible")) 
					{
						__temp_executeDef635 = false;
						this.m_visible = haxe.lang.Runtime.toBool(value);
						return value;
					}
					
					break;
				}
				
				
			}
			
			if (__temp_executeDef635) 
			{
				return super.__hx_setField(field, value, handleProperties);
			}
			 else 
			{
				throw null;
			}
			
		}
		
	}
	
	
	@Override public   java.lang.Object __hx_getField(java.lang.String field, boolean throwErrors, boolean isCheck, boolean handleProperties)
	{
		{
			boolean __temp_executeDef636 = true;
			switch (field.hashCode())
			{
				case 602652923:
				{
					if (field.equals("OnDestroy")) 
					{
						__temp_executeDef636 = false;
						return ((haxe.lang.Function) (new haxe.lang.Closure(((java.lang.Object) (this) ), haxe.lang.Runtime.toString("OnDestroy"))) );
					}
					
					break;
				}
				
				
				case 299066663:
				{
					if (field.equals("material")) 
					{
						__temp_executeDef636 = false;
						return this.get_material();
					}
					
					break;
				}
				
				
				case 1224889128:
				{
					if (field.equals("CheckCulling")) 
					{
						__temp_executeDef636 = false;
						return ((haxe.lang.Function) (new haxe.lang.Closure(((java.lang.Object) (this) ), haxe.lang.Runtime.toString("CheckCulling"))) );
					}
					
					break;
				}
				
				
				case -1623988240:
				{
					if (field.equals("get_material")) 
					{
						__temp_executeDef636 = false;
						return ((haxe.lang.Function) (new haxe.lang.Closure(((java.lang.Object) (this) ), haxe.lang.Runtime.toString("get_material"))) );
					}
					
					break;
				}
				
				
				case 424235015:
				{
					if (field.equals("UpdateCulling")) 
					{
						__temp_executeDef636 = false;
						return ((haxe.lang.Function) (new haxe.lang.Closure(((java.lang.Object) (this) ), haxe.lang.Runtime.toString("UpdateCulling"))) );
					}
					
					break;
				}
				
				
				case -74995612:
				{
					if (field.equals("set_material")) 
					{
						__temp_executeDef636 = false;
						return ((haxe.lang.Function) (new haxe.lang.Closure(((java.lang.Object) (this) ), haxe.lang.Runtime.toString("set_material"))) );
					}
					
					break;
				}
				
				
				case 1528462453:
				{
					if (field.equals("OnRender")) 
					{
						__temp_executeDef636 = false;
						return ((haxe.lang.Function) (new haxe.lang.Closure(((java.lang.Object) (this) ), haxe.lang.Runtime.toString("OnRender"))) );
					}
					
					break;
				}
				
				
				case 444448953:
				{
					if (field.equals("m_material")) 
					{
						__temp_executeDef636 = false;
						return this.m_material;
					}
					
					break;
				}
				
				
				case 312095663:
				{
					if (field.equals("OnBuild")) 
					{
						__temp_executeDef636 = false;
						return ((haxe.lang.Function) (new haxe.lang.Closure(((java.lang.Object) (this) ), haxe.lang.Runtime.toString("OnBuild"))) );
					}
					
					break;
				}
				
				
				case 466743410:
				{
					if (field.equals("visible")) 
					{
						__temp_executeDef636 = false;
						return this.get_visible();
					}
					
					break;
				}
				
				
				case 259065146:
				{
					if (field.equals("m_last_queue")) 
					{
						__temp_executeDef636 = false;
						return this.m_last_queue;
					}
					
					break;
				}
				
				
				case 543256713:
				{
					if (field.equals("get_visible")) 
					{
						__temp_executeDef636 = false;
						return ((haxe.lang.Function) (new haxe.lang.Closure(((java.lang.Object) (this) ), haxe.lang.Runtime.toString("get_visible"))) );
					}
					
					break;
				}
				
				
				case 286009348:
				{
					if (field.equals("m_has_mesh")) 
					{
						__temp_executeDef636 = false;
						return this.m_has_mesh;
					}
					
					break;
				}
				
				
				case 454676885:
				{
					if (field.equals("set_visible")) 
					{
						__temp_executeDef636 = false;
						return ((haxe.lang.Function) (new haxe.lang.Closure(((java.lang.Object) (this) ), haxe.lang.Runtime.toString("set_visible"))) );
					}
					
					break;
				}
				
				
				case -1948035677:
				{
					if (field.equals("m_culled")) 
					{
						__temp_executeDef636 = false;
						return this.m_culled;
					}
					
					break;
				}
				
				
				case -914040160:
				{
					if (field.equals("m_visible")) 
					{
						__temp_executeDef636 = false;
						return this.m_visible;
					}
					
					break;
				}
				
				
			}
			
			if (__temp_executeDef636) 
			{
				return super.__hx_getField(field, throwErrors, isCheck, handleProperties);
			}
			 else 
			{
				throw null;
			}
			
		}
		
	}
	
	
	@Override public   double __hx_getField_f(java.lang.String field, boolean throwErrors, boolean handleProperties)
	{
		{
			boolean __temp_executeDef637 = true;
			switch (field.hashCode())
			{
				case 259065146:
				{
					if (field.equals("m_last_queue")) 
					{
						__temp_executeDef637 = false;
						return ((double) (this.m_last_queue) );
					}
					
					break;
				}
				
				
			}
			
			if (__temp_executeDef637) 
			{
				return super.__hx_getField_f(field, throwErrors, handleProperties);
			}
			 else 
			{
				throw null;
			}
			
		}
		
	}
	
	
	@Override public   java.lang.Object __hx_invokeField(java.lang.String field, haxe.root.Array dynargs)
	{
		{
			int __temp_hash639 = field.hashCode();
			boolean __temp_executeDef638 = true;
			switch (__temp_hash639)
			{
				case 602652923:case 312095663:
				{
					if (( (( ( __temp_hash639 == 602652923 ) && field.equals("OnDestroy") )) || field.equals("OnBuild") )) 
					{
						__temp_executeDef638 = false;
						return haxe.lang.Runtime.slowCallField(this, field, dynargs);
					}
					
					break;
				}
				
				
				case -1623988240:
				{
					if (field.equals("get_material")) 
					{
						__temp_executeDef638 = false;
						return this.get_material();
					}
					
					break;
				}
				
				
				case 1224889128:
				{
					if (field.equals("CheckCulling")) 
					{
						__temp_executeDef638 = false;
						return this.CheckCulling();
					}
					
					break;
				}
				
				
				case -74995612:
				{
					if (field.equals("set_material")) 
					{
						__temp_executeDef638 = false;
						return this.set_material(((haxor.graphics.material.Material) (dynargs.__get(0)) ));
					}
					
					break;
				}
				
				
				case 424235015:
				{
					if (field.equals("UpdateCulling")) 
					{
						__temp_executeDef638 = false;
						this.UpdateCulling();
					}
					
					break;
				}
				
				
				case 543256713:
				{
					if (field.equals("get_visible")) 
					{
						__temp_executeDef638 = false;
						return this.get_visible();
					}
					
					break;
				}
				
				
				case 1528462453:
				{
					if (field.equals("OnRender")) 
					{
						__temp_executeDef638 = false;
						this.OnRender();
					}
					
					break;
				}
				
				
				case 454676885:
				{
					if (field.equals("set_visible")) 
					{
						__temp_executeDef638 = false;
						return this.set_visible(haxe.lang.Runtime.toBool(dynargs.__get(0)));
					}
					
					break;
				}
				
				
			}
			
			if (__temp_executeDef638) 
			{
				return super.__hx_invokeField(field, dynargs);
			}
			
		}
		
		return null;
	}
	
	
	@Override public   void __hx_getFields(haxe.root.Array<java.lang.String> baseArr)
	{
		baseArr.push("m_last_queue");
		baseArr.push("m_has_mesh");
		baseArr.push("m_culled");
		baseArr.push("m_visible");
		baseArr.push("visible");
		baseArr.push("m_material");
		baseArr.push("material");
		{
			super.__hx_getFields(baseArr);
		}
		
	}
	
	
}


